src/System.Management.Automation/engine/PSVersionInfo.cs

```diff
@@ -748,7 +882,6 @@ public override int GetHashCode()
         /// </summary>
         public static bool operator <(SemanticVersion v1, SemanticVersion v2)
         {
-            if ((object)v1 == null) throw PSTraceSource.NewArgumentException(nameof(v1));
```

Comment:
I feel we still need this check, otherwise v1.CompareTo will throw null reference exception.

Reply:
Fixed by new method Compare()